### PR TITLE
Use Select2 for improved <select> dropdowns for Forms, Landing Pages and Tags 

### PR DESCRIPTION
**Pull request by n7studios**
## Summary

[Originally noticed on the ConvertKit for WooCommerce Plugin](https://github.com/ConvertKit/tier3-issues/issues/1909), any ConvertKit account that contains a large number of Forms, Tags or Landing Pages results in a lengthly select dropdown when choosing an option:
Screenshot 2022-01-10 at 18 13 05

This isn't user friendly or intuitive when trying to find a specific Form, Tag or Landing Page.

This PR adds jQuery Select2, which replaces select dropdowns for Forms, Tags and Landing Pages to provide a scrollable selection list with search functionality.

Before:
Screenshot 2022-01-10 at 15 07 27

After:
Screenshot 2022-01-10 at 18 18 23
Screenshot 2022-01-10 at 18 18 50

## Testing

The following tests have been modified to test that Select2 works when selecting a Form, Tag or Landing Page:
- `PageFormCest`
- `PageLandingPageCest`
- `PageTagCest`
- `PluginSettingsGeneralCest`
- `PostFormCest`
- `WooCommerceProductFormCest`

## Checklist

* [x] I have [written a test](TESTING.md#writing-an-acceptance-test) and included it in this PR
* [x] I have [run all tests](TESTING.md#run-tests) and they pass
* [x] The code passes when [running the PHP CodeSniffer](TESTING.md#run-php-codesniffer)
* [x] Code meets [WordPress Coding Standards](DEVELOPMENT.md#coding-standards) for PHP, HTML, CSS and JS
* [x] [Security and Sanitization](DEVELOPMENT.md#security-and-sanitization) requirements have been followed
* [x] I have assigned a reviewer or two to review this PR (if you're not sure who to assign, we can do this step for you)

### CODE REVIEW

File: admin/section/class-convertkit-settings-base.php [hunk 1]
```diff
@@ -221,8 +225,18 @@ public function get_select_field( $name, $value = '', $options = array(), $descr
 			);
 		}
 
+		// Close <select>.
 		$html .= '</select>';
 
+		// If no description exists, just return the select field.
+		if ( ! $description ) {
```
Comment (noelherrick) [inline admin/section/class-convertkit-settings-base.php:232]: Why isn't this just an `||` with line 235?

File: tests/_support/Helper/Acceptance.php [hunk 2]
```diff
@@ -30,6 +30,27 @@ public function seeFieldContains($I, $element, $value)
...
+	 * @since 	1.9.6.4
+	 * 
+	 * @param 	AcceptanceTester 	$I
+	 * @param 	string 				$container 	Field CSS Class / ID
+	 * @param 	string 				$value 		Field Value
+	 * @param 	string 				$ariaAttributeName 	Aria Attribute Name (aria-controls|aria-owns)
+	 */
+	public function fillSelect2Field($I, $container, $value, $ariaAttributeName = 'aria-controls')
+	{
+		$fieldID = $I->grabAttributeFrom($container, 'id');
```
Comment (noelherrick) [inline tests/_support/Helper/Acceptance.php:45]: Strange formatting here.

**Review by noelherrick — APPROVED** [review]
Nice improvement!

File: tests/_support/Helper/Acceptance.php (hunk 2 above)
Comment (n7studios) [reply tests/_support/Helper/Acceptance.php:45]: Let me know what's strange with the formatting here.

Comment (noelherrick) [reply tests/_support/Helper/Acceptance.php:45]: Lines 45-46 are more indented than the rest of the function. It looks like you're mixing tabs and spaces too.

**Pull request merged**